drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/RulesExecutor.java

```diff
@@ -62,14 +63,17 @@ public long rulesCount() {
     }
 
     public CompletableFuture<Integer> executeFacts(String json) {
+        MemoryMonitorUtil.checkMemoryOccupation();
         return rulesEvaluator.executeFacts(asFactMap(json));
     }
 
     public CompletableFuture<List<Match>> processFacts(String json) {
+        MemoryMonitorUtil.checkMemoryOccupation();
         return rulesEvaluator.processFacts(asFactMap(json));
     }
 
     public CompletableFuture<List<Match>> processEvents(String json) {
+        MemoryMonitorUtil.checkMemoryOccupation();
         return rulesEvaluator.processEvents(asFactMap(json));
     }
```

Comment:
This is the place where we can check the memory before sync/async execution.

Reply:
I'm not sure that checking this synchronously and at each and every rule evaluation is a good idea. I'm afraid the overhead could be relevant and not totally justified. One possibility is to run this asynchronously as a timed task. Another alternative is doing it here and synchronously but actually doing the check (and running the gc) only once in fixed number (100?) of evaluations. What do you think?

Reply:
Thanks. I agree that we want to minimize the overhead. Probably the former (time-based check) may miss "quickly incoming many events" scenario. So I guess the latter (number-based check) is better?